modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/type/ast/ASTType.java

```diff
@@ -154,33 +171,74 @@ public TypeElement getTypeElement() {
     return typeGenerics;
   }
 
-  /**
-   * {@inheritDoc}
-   */
   @Override
-  public java.lang.reflect.Type getReflectType() {
-    return astUtils.getReflectType(this);
+  public MetadataType asMetadataType() {
+    return typeLoader.load(typeMirror).orElseThrow(RuntimeException::new);
   }
 
   /**
    * {@inheritDoc}
    */
   @Override
   public boolean isAssignableTo(Class<?> clazz) {
-    return processingEnvironment.getTypeUtils()
-        .isAssignable(processingEnvironment.getTypeUtils().erasure(typeMirror), getTypeMirrorFromClass(clazz));
+    Types types = processingEnvironment.getTypeUtils();
+    return types
+        .isAssignable(types.erasure(typeMirror), types.erasure(getTypeMirrorFromClass(clazz)));
+  }
+
+  @Override
+  public boolean isAssignableTo(Type type) {
+    if (type instanceof ASTType) {
+      return processingEnvironment.getTypeUtils()
+          .isAssignable(processingEnvironment.getTypeUtils().erasure(typeMirror),
+                        processingEnvironment.getTypeUtils().erasure(((ASTType) type).typeMirror));
+    } else if (type.getDeclaringClass().isPresent()) {
+      return processingEnvironment.getTypeUtils()
+          .isAssignable(processingEnvironment.getTypeUtils().erasure(typeMirror),
+                        getTypeMirrorFromClass(type.getDeclaringClass().get()));
+    }
+    return false;
+  }
+
+  @Override
+  public boolean isSameType(Type type) {
+    if (type instanceof ASTType) {
+      TypeMirror givenType = processingEnvironment.getTypeUtils().erasure(((ASTType) type).typeMirror);
+      TypeMirror actualType = processingEnvironment.getTypeUtils().erasure(typeMirror);
+      return processingEnvironment.getTypeUtils().isSameType(actualType, givenType);
+    } else if (type.getDeclaringClass().isPresent()) {
+      return isSameType(type.getDeclaringClass().get());
+    }
+    return false;
+  }
+
+  @Override
+  public boolean isSameType(Class<?> clazz) {
+    Types types = processingEnvironment.getTypeUtils();
+    return types
+        .isSameType(types.erasure(typeMirror), types.erasure(getTypeMirrorFromClass(clazz)));
+  }
+
+  @Override
+  public boolean isInstantiable() {
+    return getClassInformation().isInstantiable();
```

Comment:
is this the only use for getClassInformation? if that's the case, remove the method and use the `isInstantiable(asMetadataType())`

Reply:
No, also it's needed to provide the class information when building a MetadataType manually.